pymeasure/instruments/hp/hplegacyinstrument.py

```diff
@@ -0,0 +1,166 @@
+        place, decimal = 1, 0
+        while bcd > 0:
+            nibble = bcd & 0xF
+            decimal += nibble * place
+            bcd >>= 4
+            place *= 10
+        return decimal
+
+    @staticmethod
+    def _convert_to_bcd(decimal):
+        """Converts a decimal value to a bcd value
+
+        :param value: The decimal value to to pack into bcd
+        :returns: The number in bcd form
+        """
+        place, bcd = 0, 0
+        while decimal > 0:
+            nibble = decimal % 10
+            bcd += nibble << place
+            decimal //= 10
+            place += 4
+        return bcd
+
+    def __str__(self):
+        """
+        Returns a pretty formatted string showing the status of the instrument
+
+        """
+        ret_str = ""
+        for field in self._fields_:
+            ret_str = ret_str + f"{field[0]}: {hex(getattr(self, field[0]))}\n"
+
+        return ret_str
+
+    def __getattribute__(self, name):
+        val = super().__getattribute__(name)
+        if name == "fields":
+            return val
+        if name in self.fields():
+            process = super().__getattribute__('_get_process_')
+            if name in process:
+                val = process[name](val)
+
+        return val
+
+    def fields(self):
+        return [desc[0] for desc in super().__getattribute__('_fields_')]
+
+
+class HPLegacyInstrument(Instrument):
+    """
+    Class for legacy HP instruments from the era before SPCI, based on `pymeasure.Instrument`
+
+    """
+    status_desc = StatusBitsBase  # To be overriden by subclasses
+
+    def __init__(self, adapter, name, status_bitfield, **kwargs):
+        super().__init__(
+            adapter, name,
+            includeSCPI=False,
+            **kwargs,
+        )
+
+        self.status_bytes_count = ctypes.sizeof(self.status_desc)
+        self.status_bits = self.status_desc
+
+        log.info(f"Initializing {self.name}")
+
+    def write(self, command):
+        if command == "B":
+            self.adapter.connection.write("B", termination="")
+        else:
+            super().write(command)
+
+    def values(self, command, **kwargs):
+        if command == "B":
+            self.write(command)
+            return self.adapter.connection.read_raw(**kwargs)
+        else:
+            return super().values(command, **kwargs)
+
+    @property
+    def status(self):
+        """
+        Returns an object representing the current status of the unit.
+
+        """
+        self.write("B")
+        reply = bytearray(self.adapter.read_bytes(self.status_bytes_count))
+        return self.status_bits.from_buffer(reply)
+
+    def GPIB_trigger(self):
+        """
+        Initate trigger via low-level GPIB-command (aka GET - group execute trigger)
+
+        """
+        self.adapter.connection.assert_trigger()
+
+    def reset(self):
+        """
+        Initatiates a reset (like a power-on reset) of the HP3478A
+
+        """
+        self.adapter.connection.clear()
```

Comment:
Is this enough to start a reset as described in the docstring ?

Reply:
Yes it is,

Please see  pdf page 43 of the [manual](https://xdevs.com/doc/HP_Agilent_Keysight/HP%203478A%20Operations.pdf).

The HP3437 & HP663xA react in the same way